### ISSUE TITLE
Math formulas on a lot of pages are invisible

### ISSUE THREAD

**Issue by Fabian42** [darkreader/darkreader#1273]
Similar to #649. Might be caused by #356.

There seems to be a certain way of embedding Math formulas that many websites use that doesn't work well with Dark Reader. Often it's completely invisible (black font on black background), sometimes a slight shadow is visible, but usually not enough to properly read the formula.

Some examples

* https://en.wikipedia.org/wiki/Invertible_matrix
![image](https://user-images.githubusercontent.com/10481506/56844922-f2e6e480-68b8-11e9-84ff-60979dcdd61a.png)
Here the "Filter+" mode fixes it, but in the default it's invisible.

* https://elearning.physik.uni-frankfurt.de/data/FB13-PhysikOnline/lm_data/lm_281/modul_2/teil_8/node48.html
![image](https://user-images.githubusercontent.com/10481506/56844929-127e0d00-68b9-11e9-9234-995238e2de42.png)
Here all four modes look similarly bad. The only way to read the formula is to disable Dark Reader for this site.

There are also cases where all four modes have completely invisible formulas, I just didn't find any right now.

**Comment by NewbieToEverything** [comment]
Similar issue encountered, I use Wikipedia as mathematical reading material quite often and this bugs me very much. Hope there exists a fix.

**Comment by Just-Harry** [comment]
Perhaps a configuration setting could be put in place for this?
To not modify the `background-color` of images, unless the alpha value of the computed `background-color` of the image is below a certain value, in which case set the `background-color` of the image to the original (non-inverted) `background-color` of its parent-element.

---

As a personal ad-hoc solution, I'm using a bookmarklet which toggles the `background-color` of all images between white, and their original `background-color`:

`javascript:(/* Licence: CC0 */ function () {for (let img of document.body.getElementsByTagName('img')) {img.style.setProperty('background-color', img.style.getPropertyValue('background-color') === 'white' ? 'initial' : 'white', 'important');}})();`

Although, one could also use Dark Reader's dev tools to make the `background-color` of all images white by placing the following rule in the CSS section of the very first `*` URL selector.

```
img {
    background-color: white !important;
}
```

**Comment by rkanati** [comment]
Bizarrely, removing the math-related classes under `INVERT` for `wikipedia.org` in the style editor fixes this for me - can someone else try?

**Comment by Just-Harry** [comment]
> 
> 
> Bizarrely, removing the math-related classes under `INVERT` for `wikipedia.org` in the style editor fixes this for me - can someone else try?

Removing `.mwe-math-element` and `.mwe-math-fallback-image-inline` from under `INVERT` makes the formulae images even harder to see, for me.

**Comment by rkanati** [comment]
Strange. This may be an interaction with Firefox's colour preferences, in my case; I don't understand this well enough to say.

**Comment by iamkroot** [comment]
> Bizarrely, removing the math-related classes under `INVERT` for `wikipedia.org` in the style editor fixes this for me - can someone else try?

Removing the `.mwe-math-element` seems to be working in my case too with Dynamic mode. I'm running Vivaldi browser, so it should be the same for all other Chromium browsers.

**Comment by JanosSarkoezi** [comment]
A possible solution for
- https://en.wikipedia.org/wiki/Invertible_matrix
  - open darkreader developer tool
  - search for wikipedia.org
  - replace content between ===== with
```
wikipedia.org

INVERT
.mwe-math-element
.mw-wiki-logo
.central-textlogo__image
.svg-Wikimedia-logo_black

CSS
.mwe-math-fallback-image-inline {
     color: FFF !important;
} 
```

The main idea on wikipedia.org is to remove *mwe-math-fallback-image-inline* AND add css section to content.

- https://elearning.physik.uni-frankfurt.de/data/FB13-PhysikOnline/lm_data/lm_281/modul_2/teil_8/node48.html
  - open darkreader developer tool
  - add following content before akinator.com
```

elearning.physik.uni-frankfurt.de

INVERT
img

================================
```

At first sight it looks as if the structure of the file has a main section and then subsections. The sections are separated by a separator of the form ====. The subsections included a part of a URI, an INVERT area and a CSS area. With this solution I only added a part of a URI, an INVERT area and a separator.

After saving the file, the program arranges the contents of the data according to the ABC of the URI. If anything goes wrong, just press the reset button and everything will be fine.

Hope this helps solve the problem. :)

**Comment by Manu99it** [comment]
Actually I tried to write img.mwe-math-element instead of .mwe-math-element and seems to work

**Comment by tylerliu1998** [comment]
Another solution is to use Wikiwand or Stylish extension in the meantime if you do not want to fix anything in Darkreader.

**Comment by jbriales** [comment]
@tylerliu1998 did you find any dark theme in Stylish that inverts math color as well?

**Comment by jbriales** [comment]
Wikiwand worked great for me. DarkReader inverts math properly there plus the interface in Wikiwand feels so much nicer. Fastest solution to this issue I found :)

**Comment by tylerliu1998** [comment]
> @tylerliu1998 did you find any dark theme in Stylish that inverts math color as well?

I also use Wikiwand. Stylish might be an alternative solution but I have not tried since Wikiwand is enough.

**Comment by Just-Harry** [comment]
I'd recommend using [Stylus](https://github.com/openstyles/stylus) over Stylish. Stylus being a fork of the last open-source release of Stylish, which has since been made closed-source, [and was found to contain spyware (or at the very least, over-reaching telemetry)](https://www.theregister.co.uk/2018/07/05/browsers_pull_stylish_but_invasive_browser_extension/).

**Comment by Gusted** [comment]
Closed by #1805
